pkg/controller/vsphere/reconciler.go

```diff
@@ -217,12 +232,17 @@ func (r *Reconciler) delete() error {
 		}
 		if moTask != nil {
 			if taskIsFinished, err := taskIsFinished(moTask); err != nil {
-				metrics.RegisterFailedInstanceDelete(&metrics.MachineLabels{
-					Name:      r.machine.Name,
-					Namespace: r.machine.Namespace,
-					Reason:    "Task finished with error",
-				})
-				return err
+				// Check if latest task is not a task for vm cloning
+				if taskIsFinished && moTask.Info.DescriptionId != cloneVmTaskDescriptionId {
+					metrics.RegisterFailedInstanceDelete(&metrics.MachineLabels{
+						Name:      r.machine.Name,
+						Namespace: r.machine.Namespace,
+						Reason:    "Task finished with error",
+					})
+					return err
+				} else {
+					return err
```

Comment:
Can we add some extra descriptions to these errors so that we can tell from the logging which route it took here?